lib/util/search.js

```diff
@@ -19,16 +19,38 @@ function SearchClient(crowi, esUri) {
   this.client = new elasticsearch.Client({
     host: this.host,
     requestTimeout: 5000,
+    //log: 'trace',
   });
 
   this.registerUpdateEvent();
-
   this.mappingFile = crowi.resourceDir + 'search/mappings.json';
+
+  this.serverInfo = {};
+
+  Promise
+    .resolve(this.client.info())
+    .then((info) => {
+      this.serverInfo = info;
+    })
+    .catch((err) => {
+      throw new Error();
+    });
 }
 
-SearchClient.prototype.checkESVersion = function() {
-  // TODO
-};
+SearchClient.prototype.isSupportedESVersion = function() {
+  var crowiVersion = this.crowi.getParsedVersion();
+  var ESVersion    = this.crowi.getParsedVersion(this.serverInfo.version.number);
+
+  if (crowiVersion.major === 1 && crowiVersion.minor >= 6) {
+    if (ESVersion.major >= 5) {
+      return true;
+    }
+  }
+
+  console.log('WARNING: Detect unsupported Elasticsearch version: ' + this.serverInfo.version.number);
```

Comment:
When unsupported es version detected, Crowi may be good to shutdown.
(Because the operator clearly specified the ES's URI on launching Crowi)